Users/Tutorial.rst

```diff
@@ -292,10 +303,26 @@ After the results we've already seen, we'll see a new informational one
 which informs us that we have a TODO in our code.
 
 Did you note that we didn't specify which files to check this time? This
-is because all settings, including ``files = src/*.c``, from the Default
-section are already available in every other section implicitly. Thus
-the default section is a good point to set things like logging and
-output settings or specifying a default set of files to check.
+is because all settings, including ``files = src/*.c``, from the ``all``
+section (previously called ``cli``) have been inherited in the new
+``TODOS`` section that we just added.
+
+You can make a section inherit from any previously defined section using
+this syntax:
+
+::
+
+    [parentSection.childSection]
+
+.. note::
+
+    ``cli`` is an internally reserved section name. All of its settings
+    are implicitly inherited to every other section by default. It is
+    because of this implicit inheritance feature that we are adviced to
+    rename the ``cli`` section to something else. Doing so will save us
+    from having unexpected values of ``cli`` being implicitly inherited
+    into our sections. We suggest that you change its name to ``all``.
```

Comment:
Ah :3 Regarding ☝️  https://github.com/coala/documentation/pull/537/commits/edf0dae63b7b8e93eb48e088b609dd3a39b6e800#r170088538

So this is what should come earlier, in the `Configuration Files - coafiles`-paragraph.

Reply:
IMHO this note can only be understood by the reader if he/she is already aware of the inheritance features of coala's sections. Putting this note before we are done explaining inheritance does not sound like a good explanation.

I suggest that we put a small line at the end of that earlier note saying- 
> Read the next section to understand what it [*the warning message*] means and how you can avoid it.